### DISCUSSION TITLE
Modifications by Yuancheng Xu to data_handling.py & landcoverType.py

### DISCUSSION THREAD

**Discussion by YuanchengXu2718** [iiasa/CWatM#123] [General]
I made some minor modifications to the CWatM and hope these help. For 1, I understand that using a convolute of different spatial extend and then correcting this in CWatM is prone to errors. My intention in mentioning this is to underscore that **no warning does not mean everything is going well**. For 2, it is just a cosmetic improvement and does not impact the results.
1.	In data_handling.py, function readnetcdf2, at around Line 1100. It might be better if each cutmap is determined by the input file it belongs to. Originally, the cutmap is defined by the LDD map. Unfortunately, it takes a long time for regional studies to produce input files with the same spatial coverage. What’s worse, if the **spatial resolutions of the input files are the same as the spatial resolution of Maskmap and LDD**, the model works without warning. It simply **uses the top left corners of the input files** which has a larger spatial coverage than the Maskmap.
![image](https://github.com/iiasa/CWatM/assets/134045269/5c432214-4dae-456d-b26c-b29bb24f8a94)
2.	In landcoverType.py, function initial, at around Line 500. The initialization of the soil water extent should not be inside the loop around Line 438.
![image](https://github.com/iiasa/CWatM/assets/134045269/d2fd613f-6cd6-4e54-bda4-471b9131a7fa)

**Comment by mikhailsmilovic** [comment]
Guten Tag Yuancheng Xu, 

For point 1: The concept is that you produce a mask map, which can be the whole area or a part of it. This mask map will be used to run the model. Producing input maps for an entire region takes some time, but this time is well invested in ensuring a homogenous set of input maps for that region. Using a convolution of different spatial extents and correcting this in CWatM is prone to errors. 

_Our advice:_ Spend some time making an organised set of maps for the region. Same resolution, same extent and then use a mask map to run the part of the region you are interested

You can comfortably include changes by forking the repository and committing the changes. Then, you can submit a pull request to merge your changes with the main repository. This [video](https://www.youtube.com/watch?v=tjaK-Xx5hCs&list=PLyT8dd_rWLawkTUNNXIIhoTxm_fGkI9Vf&index=6&ab_channel=MikhailSmilovic) provides a detailed example of this process. We look forward to your contributions!
